qiskit_finance/applications/optimization/portfolio_optimization.py

```diff
@@ -0,0 +1,171 @@
+        quad = mdl.quad_matrix_sum(self._covariances, x)
+        linear = np.dot(self._expected_returns, x)
+        mdl.minimize(self._risk_factor * quad - linear)
+        mdl.add_constraint(mdl.sum(x[i] for i in range(num_assets)) == self._budget)
+        op = QuadraticProgram()
+        op.from_docplex(mdl)
+        return op
+
+    def portfolio_expected_value(self, result: Union[OptimizationResult, np.ndarray]) -> float:
+        """Returns the portfolio expected value based on the result.
+
+        Args:
+            result: The calculated result of the problem
+
+        Returns:
+            The portfolio expected value
+        """
+        x = self._result_to_x(result)
+        return np.dot(self._expected_returns, x)
+
+    def portfolio_variance(self, result: Union[OptimizationResult, np.ndarray]) -> float:
+        """Returns the portfolio variance based on the result
+
+        Args:
+            result: The calculated result of the problem
+
+        Returns:
+            The portfolio variance
+        """
+        x = self._result_to_x(result)
+        return np.dot(x, np.dot(self._covariances, x))
+
+    def interpret(self, result: Union[OptimizationResult, np.ndarray]) -> List[int]:
+        """Interpret a result as a list of asset indices
+
+        Args:
+            result: The calculated result of the problem
+
+        Returns:
+            The list of asset indices whose corresponding variable is 1
+        """
+        x = self._result_to_x(result)
+        return [i for i, x_i in enumerate(x) if x_i]
+
+    @property
+    def expected_returns(self) -> np.ndarray:
+        """Getter of expected_returns
+
+        Returns:
+            The expected returns for the assets.
+        """
+        return self._expected_returns
+
+    @expected_returns.setter
+    def expected_returns(self, expected_returns: np.ndarray) -> None:
+        """Setter of expected_returns
+
+        Args:
+            expected_returns: The expected returns for the assets.
+        """
+        self._expected_returns = expected_returns
+
+    @property
+    def covariances(self) -> np.ndarray:
+        """Getter of covariances
+
+        Returns:
+            The covariances between the assets.
+        """
+        return self._covariances
+
+    @covariances.setter
+    def covariances(self, covariances: np.ndarray) -> None:
+        """Setter of covariances
+
+        Args:
+            covariances: The covariances between the assets.
+        """
+        self._covariances = covariances
+
+    @property
+    def risk_factor(self) -> float:
+        """Getter of risk_factor
+
+        Returns:
+            The risk appetite of the decision maker.
+        """
+        return self._risk_factor
+
+    @risk_factor.setter
+    def risk_factor(self, risk_factor: float) -> None:
+        """Setter of risk_factor
+
+        Args:
+            risk_factor: The risk appetite of the decision maker.
```

Comment:
Maybe the text should say `risk factor` I know people talk about whether you might have an appetite for risk but maybe it might not be so clear to some. Anyway its pretty minor since its on the setter/getter which should be more obvious so I approved anyway